First pass at setting up one-command training with skypilot.

alxmrs opened the pull request:
Wrote entrypoint for performing training runs with skypilot. The `sky` command is documented within the `train.sky.yaml` file and contributing guide.

I did not successfully do a full training run, but that is because I hit data engineering errors (no static_data!).

<details>
<summary>(Latest error trace related to data engineering)</summary

Ultimately, hitting this error seems like a good sign because the error is in our code, not in the provisioning system.
```
AssertionError: : Static data variable sea_surface_fraction not found in dataStatic data variable sea_surface_fraction not found in data
```

</details<

alxmrs committed:
First pass at setting up one-command training with skypilot.

alxmrs committed:
Fixed config path, pointing to new bucket. Added `uv` setup.

alxmrs committed:
Adding a data config for freshly cloned remote data.

alxmrs committed:
Creating a config for local, but managed by sky, datasets.

alxmrs committed:
Don't need remote data loading after all.

alxmrs committed:
Prevent accidentally checking in .env file.

alxmrs committed:
Added `null` to force users to specify specific variables. Also, fixed use of envvars.

alxmrs committed:
Made the data config... configurable.

alxmrs committed:
Making sure to ignore files during sky's workdir sync.

alxmrs committed:
Config fixes. Still, `secrets:` doesn't work... :hmm:

alxmrs committed:
Corrected a bash var

alxmrs committed:
Making more progress! Job gets as far as syncing the workdir, but then fails to mount storage with rsync due to lack of credentials.

alxmrs committed:
Trying to add more directories to be ignored during workdir sync!

alxmrs committed:
Added a helpful note to the skyignore. Copied gitignore into skyignore.

alxmrs committed:
Moved to skypilot nightly; updated docstrings in train entrypoint.

alxmrs committed:
Adding local .git/ dir to skyignore

alxmrs committed:
Yo @jder, what's up

alxmrs committed:
Workaround for passing in the data root (fingers crossed)

alxmrs committed:
data root can't be null.

alxmrs committed:
Added the static data vars; hit data eng error.

alxmrs committed:
Added comment.

alxmrs commented on train.sky.yaml:
```diff
@@ -0,0 +1,84 @@
+  # https://docs.skypilot.co/en/latest/cloud-setup/cloud-permissions/aws.html#dedicated-skypilot-iam-user
+
+  CONFIG: null
+  EXPERIMENT_NAME: null
```
Question for the reviewer: which config system do we want this to be driven from? the .env / CLI, or the config file?

jder replied:
My vote would be to use the sky Python SDK and drive it from our existing config system but if that's a pain we can use env vars here.

alxmrs replied:
I like this idea! I think it's a bigger lift than I want to invest for now. I'll let the experiment name come from the config, but have data and secrets be driven by this YAML's env vars. I think this makes sense bc for the most part, data location won't change as fast as the rest of the configuration.

alxmrs committed:
Added a contributing guide note.

alxmrs commented on train.sky.yaml:
```diff
@@ -0,0 +1,84 @@
+      exit 1
+  fi
+
+  uv run torchrun \
```
`uv` comes pre-installed on skypilot machines.

jder commented on .skyignore:
```diff
@@ -0,0 +1,55 @@
+*.nc
+# Copied from .gitignore on 2025-07-15
+# .skyignore overrides the .gitignore default!
+# See - https://docs.skypilot.co/en/latest/examples/syncing-code-artifacts.html#exclude-uploading-files
```
Might be nicer to let it use the existing `.gitignore`, is there anything we can't put in there which we need to prevent sky from uploading?

alxmrs replied:
I wanted to ignore files we use for maintaining the project (`scripts/`, `tests/`, `.github/`, etc.) but are not needed for training runs. These are not too consequential to upload, so happy to just use the .gitignore. WDYT?

jder replied:
Yeah, my vote would be to let it upload those and have a single list in .gitignore, but this is a 🐑-level opinion.

alxmrs replied:
Given Ryan's recent comment related to personal/global .gitignore configs, I think that opinion is popular. Will go with just using .gitignore.

jder commented on configs/data/om4_sky_local.yaml:
```diff
@@ -0,0 +1,17 @@
+
+data_stds_location:
+  type: local
+  path: /data/OM4_means.zarr
```
See https://github.com/Open-Athena/Ocean_Emulator/pull/309, these should be eg `data_location: "OM4.zarr"`

alxmrs replied:
Thanks again for this fix!

jder commented on train.sky.yaml:
```diff
@@ -0,0 +1,84 @@
+#  disk_size: 1_000 # in GiBs
+
+# FIXME: Remember to pass environment variables with `--env` or `--env-file`
+# FIXME: All `null` values must be specified in order for this script to run!
```
What would we fix for these?

jder commented on train.sky.yaml:
```diff
@@ -0,0 +1,84 @@
+  # Secrets
+  # As of v0.9.3, `sky` supports passing in secrets via a dotenv file! So, you can specify these with `--env-file`.
+  # To pass in secrets, we recommend setting environment variable and then ONLY passing in the name via `--env`
+  WANDB_API_KEY: null  # e.g. `--env WANDB_API_KEY`
```
It looks like secrets should go in a `secrets` dict and are handled differently (eg obscured from UI)? https://docs.skypilot.co/en/latest/running-jobs/environment-variables.html#user-specified-environment-variables-and-secrets and their recommendation is to set them in your env and use `--secret WANDB_API_KEY`

alxmrs replied:
This changed a lot in recent versions! It used to go in `--secrets`, but in the 0.9.3 and nightly versions, this option is combined.

jder commented on train.sky.yaml:
```diff
@@ -0,0 +1,84 @@
+    --experiment.name=${EXPERIMENT_NAME} \
+    --experiment.base_output_dir=/outputs  \
+    --experiment.wandb.mode=online \
+    --samudra.checkpointing=all
```
🐑 If we want to keep using the sky yaml/cli, we should maybe consider turning on the env var resolution stuff in the config system so we don't need to thread things through here. (ie setting EXPERIMENT_NAME will be enough). Others here like data_root and base_output_dir are maybe worth moving somewhere that we can have a sky-specific config file for?

jder reviewed (approved):
Awesome stuff! I'm sure we'll iterate on the exact structure here as we figure out what we want to do with this. We should resolve the secrets thing before merging but otherwise I think we can merge + iterate from here.

alxmrs committed:
Implementing review feedback:
- Making use of recent absolute path for local paths
- Changing fixme to note
- Experiment name should come from the config.

alxmrs committed:
Favoring .gitignore over .skyignore.
